src/main/java/com/google/devtools/build/lib/analysis/AnalysisOptions.java

```diff
@@ -39,6 +39,17 @@ public class AnalysisOptions extends OptionsBase {
   )
   public boolean discardAnalysisCache;
 
+  @Option(
+      name = "fail_on_analysis_cache_discard",
```

Comment:
In the bazel-discuss thread @alexeagle suggested `disallow_analysis_cache_discards`, open to other suggestions.